sdk/appconfiguration/Azure.Data.AppConfiguration/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,165 @@
+
+* [General troubleshooting](#general-troubleshooting)
+  * [Enable client logging](#enable-client-logging)
+* [Troubleshooting authentication issues](#troubleshooting-authentication-issues)
+  * [AuthenticationFailedException](#authenticationfailedexception)
+  * [CredentialUnavailableException](#credentialunavailableexception)
+  * [Permission Issues](#permission-issues)
+
+## General troubleshooting
+
+All app configuration service operations will throw a [RequestFailedException](https://docs.microsoft.com/dotnet/api/azure.requestfailedexception?view=azure-dotnet) on failure.
+
+When you interact with the library, errors returned by the service correspond to the same HTTP status codes returned for [REST API](https://learn.microsoft.com/azure/azure-app-configuration/rest-api/) requests. 
+
+For example, if you try to retrieve a Configuration Setting that doesn't exist in your Configuration Store, a `404` error is returned, indicating `Not Found`.
+
+Here's an example of how to catch an exception using synchronous method:
+
+```C# Snippet:ThrowNotFoundErrorSync
+string connectionString = "<connection_string>";
+var client = new ConfigurationClient(connectionString);
+try
+{
+    ConfigurationSetting setting = client.GetConfigurationSetting("nonexistent_key");
+}
+catch (RequestFailedException ex) when (ex.Status == 404)
+{
+    Console.WriteLine("Key wasn't found.");
+}
+```
+
+Here's an example of how to catch an exception using asynchronous method:
+
+```C# Snippet:ThrowNotFoundErrorAsync
+string connectionString = "<connection_string>";
+var client = new ConfigurationClient(connectionString);
+
+try
+{
+    ConfigurationSetting setting = await client.GetConfigurationSettingAsync("nonexistent_key");
+}
+catch (RequestFailedException ex) when (ex.Status == 404)
+{
+    Console.WriteLine("Key wasn't found.");
+}
+```
+
+You will notice that additional information is logged, like the Client Request ID of the operation.
+
+```shell
+Message: Azure.RequestFailedException : StatusCode: 404, ReasonPhrase: 'Not Found', Version: 1.1, Content: System.Net.Http.NoWriteNoSeekStreamContent, Headers:
+{
+  Connection: keep-alive
+  Date: Thu, 11 Apr 2019 00:16:57 GMT
+  Server: nginx/1.13.9
+  x-ms-client-request-id: cc49570c-9143-411e-a6c8-3287dd114034
+  x-ms-request-id: 2ad025f7-1fe8-4da0-8648-8290e774ed61
+  x-ms-correlation-request-id: 2ad025f7-1fe8-4da0-8648-8290e774ed61
+  Strict-Transport-Security: max-age=15724800; includeSubDomains;
+  Content-Length: 0
+}
+```
+
+### Enable client logging
+
+To troubleshoot issues with the library, first enable logging to monitor the behavior of the application. The errors and warnings in the logs generally provide useful insights into what went wrong and sometimes include corrective actions to fix issues.
+
+This library uses the standard [logging](https://docs.microsoft.com/dotnet/azure/sdk/logging) library. Basic information about HTTP sessions, such as URLs and headers, is logged at the `INFO` level.
```

Comment:
We should consider rewording this.  There's absolutely nothing standard about ETW and it's not widely used or understood outside of Microsoft.    The link is good.. we should keep that.